plugin/converter/jsonnet/jsonnet.go

```diff
@@ -125,7 +128,18 @@ func Parse(req *core.ConvertArgs, fileService core.FileService, limit int, templ
 	if len(templateData) != 0 {
 		for k, v := range templateData {
 			key := fmt.Sprintf("input." + k)
-			val := fmt.Sprint(v)
+			var val string
+			switch reflect.TypeOf(v).Kind() {
+			case reflect.Slice, reflect.Map:
+				yml, _ := yaml.Marshal(v)
+				data, err := json.YAMLToJSON(yml)
+				if err != nil {
```

Comment:
do we need to marshal to yaml and then json? or can we just marshal directly to json, and skip marshaling to yaml?

```diff
data, _ := json.Marshal(v)
val = string(data)
```

Reply:
Marshalling directly to json gives the `json: unsupported type: map[interface {}]interface {}` error, which I don't quite know how to work around. I've looked up the code for Docker runner and found that [this exact approach](https://github.com/drone-runners/drone-runner-docker/blob/master/internal/encoder/encoder.go#L39) was used for plugin parameter serialization, but I didn't bother to copy over the whole `encoder` package for my case. Please let me know if you have some better approach in mind

Reply:
Also, checking the sources of jsonnet-go lib reveals that [std.ParseYaml](https://github.com/google/go-jsonnet/blob/v0.18.0/builtins.go#L1198) too just converts the incoming yaml to json, then the behavior is the same as in [std.parseJson](https://github.com/google/go-jsonnet/blob/v0.18.0/builtins.go#L1184). So, if you trust jsonnet's yaml conversion enough, we could just stick with yaml and let jsonnet do the yaml-to-json stuff for us